### PR TITLE
Feature: create providers

### PR DESCRIPTION
**Pull request by teddyjfpender**
# Feature: create providers
Quite a large PR. 

Goal: standardise how we query for data beginning with a `ChainHistoryProvider`.

This adds a few more packages, including `@palladxyz/mina-core`,  `@palladxyz/mina-graphql`, and `@palladxyz/util`.

What does this provider do?
- `healthCheck()`: This method performs a basic health check of the GraphQL API. It sends a request to the API with a GraphQL query that retrieves a list of types available in the schema. If the API returns a response with at least one type in the schema, the method returns `{ ok: true }`. If there's an error during the process or the response doesn't include any types, the method returns `{ ok: false }` with an appropriate error message.

- `transactionsByAddresses(args: TransactionsByAddressesArgs)`: This method retrieves transactions related to a given list of addresses. It sends a request to the GraphQL API with a query that asks for transactions either from or to the addresses. It includes pagination parameters (`startAt` and limit) to specify which subset of results to return. The method returns a `Paginated<Mina.Transaction>` object, which includes an array of transactions (`pageResults`) and a count of total results (`totalResultCount`).

- `transactionsByHashes(args: TransactionsByIdsArgs)`: This method retrieves transactions by their hash values. It sends individual requests to the GraphQL API for each provided transaction hash, using a query that retrieves the details of a transaction given its hash. The method returns an array of `Mina.Transaction` objects.

Other providers added include:
- `AccountInfoProvider`
- `TxSubmitProvider`
- `TxStatusProvider`

It also includes the first `wallet` implementation.

I took a lot of inspiration from the great engineering work done by the team on `cardano-js-sdk` ❤️

### CODE REVIEW

**Comment by teddyjfpender** [comment]
There is a failing test in the `TxSubmitProvider` due to a bug in transaction signing.

File: packages/mina-core/src/Mina/types/Transaction.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { InvalidStringError, OpaqueString } from '@palladxyz/util'
+import {
+  PublicKey,
+  UInt32,
+  UInt64
+} from 'mina-signer/dist/node/mina-signer/src/TSTypes'
+export type Transaction = {
```
[the remark this answers is not in the document]
Comment (teddyjfpender) [reply packages/mina-core/src/Mina/types/Transaction.ts:7]: Should I look to refactor this later the core types later?

File: packages/mina-core/src/Providers/TxStatusProvider/types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Provider } from '../Provider'
+
+export type TxStatusArgs = {
+  ID: string
+}
+
+type TxStatus = {
+  status: 'PENDING' | 'INCLUDED' | 'UNKNOWN' | 'FAILED'
+}
```
Comment (mrcnk) [inline packages/mina-core/src/Providers/TxStatusProvider/types.ts:9]:
```suggestion
enum TxStatus = {
  PENDING = 'PENDING',
  INCLUDED = 'INCLUDED',
  UNKNOWN = 'UNKNOWN',
  FAILED = 'FAILED'
}
```

File: packages/mina-graphql/src/Providers/TxSubmit/mutations.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+              fee: $fee,
+              amount: $amount,
+              to: $to,
+              from: $from,
+              memo: $memo,
+              nonce: $nonce,
+              validUntil: $validUntil
+            }, 
+            signature: {rawSignature: $rawSignature}) {
+            payment {
```
Comment (mrcnk) [inline packages/mina-graphql/src/Providers/TxSubmit/mutations.ts:18]: We can try using GraphQL's fragments for the repeating parts of queries and mutations I think.
https://blog.logrocket.com/graphql-fragments-explained/
For now it's good enough, but let's create a ticket for this debt to create reusable pieces.

**Review by mrcnk — APPROVED** [review]
Beautifully done 😎

**Review by mrcnk — APPROVED** [review]

**Pull request merged**